modules/calib3d/src/ap3p.h

```diff
@@ -0,0 +1,76 @@
+#ifndef P3P_P3P_H
+#define P3P_P3P_H
+
+#include "precomp.hpp"
+
+class ap3p {
+ private:
+  inline void polishQuarticRoots(const double *coeffs, double *roots);
+
+  inline void vect_cross(const double *a, const double *b, double *result);
+
+  inline double vect_dot(const double *a, const double *b);
+
+  inline double vect_norm(const double *a);
+
+  inline void vect_scale(const double s, const double *a, double *result);
+
+  inline void vect_sub(const double *a, const double *b, double *result);
+
+  inline void vect_divide(const double *a, const double d, double *result);
+
+  inline void mat_mult(const double a[3][3], const double b[3][3], double result[3][3]);
+
+  void solveQuartic(const double *factors, double *realRoots);
```

Comment:
This functionality should not be implemented as a "class methods". Please place this code in the `.cpp` file under anonymous namespace (`namespace { ... }`) or as a "static" functions.